[go1.15] Update to go1.15.5

justaugustus opened the pull request:
**What type of PR is this?**

/kind feature
/area dependency

**What this PR does / why we need it**:

- Update to go1.15.5
- build: Update to k/repo-infra@v0.1.3 (supports go1.15.5)

Tracking issue: https://github.com/kubernetes/release/issues/1651
Slack thread: https://kubernetes.slack.com/archives/C2C40FMNF/p1605129514273400
cc: @kubernetes/release-engineering 

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

/hold this still needs a go-runner update

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
- Kubernetes is now built using go1.15.5
- build: Update to k/repo-infra@v0.1.2 (supports go1.15.5)
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

justaugustus committed:
[go1.15] build: Update to k/repo-infra@v0.1.3 (supports go1.15.5)

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus committed:
[go1.15] hack/tools: Update to k/repo-infra@v0.1.3 (supports go1.15.5)

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus committed:
[go1.15] Update to go1.15.5

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus committed:
[go1.15] staging/publishing: Set default go version to go1.15.5

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
/triage accepted
/priority important-soon

justaugustus commented:
/test pull-kubernetes-bazel-build

justaugustus commented:
/test pull-kubernetes-node-e2e

justaugustus commented:
```console
ERROR: /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/external/io_bazel_rules_go/BUILD.bazel:29:1: GoStdlib external/io_bazel_rules_go/stdlib_/pkg failed (Exit 34). Note: Remote connection/protocol failed with: execution failed catastrophically UNAVAILABLE: HTTP/2 error code: NO_ERROR 
```

:this-is-fine:
/test pull-kubernetes-bazel-build

saschagrunert reviewed (approved):
/lgtm

aojea commented:
/assign @liggitt

dims commented:
/approve (for /hack)

liggitt commented on build/dependencies.yaml:
```diff
@@ -99,22 +99,22 @@ dependencies:
+    version: v1.15.5-1
     refPaths:
     - path: build/build-image/cross/VERSION
-    - path: test/images/sample-apiserver/Makefile
```
curious why this got dropped

justaugustus replied:
@liggitt -- the sample apiserver uses a different image now and the reference no longer exists: https://github.com/kubernetes/kubernetes/blob/0c973409051faa8926971cfb05b967a2d88c5ca8/test/images/sample-apiserver/BASEIMAGE#L1

liggitt replied:
ack

liggitt replied:
actually, is that true? I still see `k8s.gcr.io/build-image/kube-cross:v1.15.2-1` at https://github.com/kubernetes/kubernetes/blob/master/test/images/sample-apiserver/Makefile#L27

liggitt reviewed (commented):
one question, lgtm otherwise

MonzElmasry commented:
@liggitt is this one approved?

liggitt commented:
> @liggitt is this one approved?

https://github.com/kubernetes/kubernetes/pull/95776#discussion_r525391271 is still outstanding

justaugustus committed:
[go1.15] Use go-runner:buster-v2.2.2 image (built on go1.15.5)

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
/hold cancel

Fixed https://github.com/kubernetes/kubernetes/pull/95776#discussion_r525391271 and added in the go-runner go1.15.5 image (`go-runner:buster-v2.2.2`) from https://github.com/kubernetes/release/pull/1780.

liggitt commented:
/lgtm
/approve

justaugustus commented:
/test pull-kubernetes-conformance-kind-ga-only-parallel
